File: backend/app/schemas/documento.py
"""Schemas Pydantic — Documento."""
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.schemas.estructura import EstructuraNode


class DocumentoBase(BaseModel):
    nombre: str
    tipo: str = "guia"


class DocumentoOut(DocumentoBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    estado: str
    paginas: int | None = None
    fecha_subida: datetime


class DocumentoUploadResponse(BaseModel):
    documento: DocumentoOut
    estructura: list[EstructuraNode]

File: frontend/types/documento.ts
export interface EstructuraNode {
  titulo: string;
  jerarquia: number;
  orden: number;
  pagina_inicio: number | null;
  hijos?: EstructuraNode[];
}

export interface Documento {
  id: string;
  nombre: string;
  tipo: string;
  estado: string;
  paginas: number | null;
  fecha_subida: string;
}

export interface UploadResponse {
  documento: Documento;
  estructura: EstructuraNode[];
}
